app/services/foreman_deprecation.rb

```diff
@@ -0,0 +1,24 @@
+class ForemanDeprecation
+  def self.deprecation_warning(moved_to_controller, moved_to_method, foreman_version_deadline, additional_info = nil,callstack = caller)
+    raise Foreman::Exception, "Invalid version format, please enter in x.y.z" unless foreman_version_deadline.to_s.match(/^\d[.]\d\d?[.]\d$/)
+    msg = "You are using deprecated behavior, it will be removed in version #{foreman_version_deadline}, use #{moved_to_controller}##{moved_to_method} instead.\n"
+    msg += additional_info unless additional_info.nil?
+    ActiveSupport::Deprecation.warn(msg,callstack)
```

Comment:
I sent the warn method callstack parameter which is the caller from the previous call, so it should log right I think.